src/lib/Name.hs

```diff
@@ -323,6 +330,18 @@ data Nest (binder::B) (n::S) (l::S) where
   Nest  :: binder n h -> Nest binder h l -> Nest binder n l
   Empty ::                                  Nest binder n n
 
+data RNest (binder::B) (n::S) (l::S) where
+  RNest  :: RNest binder n h -> binder h l -> RNest binder n l
+  REmpty ::                                   RNest binder n n
+
+unrevNest :: RNest b n l -> Nest b n l
```

Comment:
I find the name `unrevNest` unintuitive.  Maybe call this `reverseNest` (though that's ambiguous with going from `Nest` to `RNest`), or `unRNest` to evoke the type more precisely?  Or `rNestToNest`?

Reply:
Renamed to `unRNest`.